django_celery_extensions/task.py

```diff
@@ -276,10 +310,32 @@ def _start(self, *args, **kwargs):
         return self.run(*args, **kwargs)
 
     def _get_unique_key(self, task_args, task_kwargs):
-        return self.unique_key_generator(task_args, task_kwargs) if self.unique else None
+        return self.unique_key_generator(
+            settings.UNIQUE_TASK_KEY_PREFIX, task_args, task_kwargs
+        ) if self.unique else None
+
+    def _get_ignore_task_after_success_key(self, task_args, task_kwargs):
+        return (
+            self.unique_key_generator(
+                settings.IGNORE_TASK_AFTER_SUCCESS_KEY_PREFIX, task_args, task_kwargs
+            )
+            if self.ignore_task_after_success_timedelta else None
+        )
+
+    def _ignore_task_after_success(self, key):
+        return key and cache.get(key)
 
-    def _clear_unique_key(self, task_args, task_kwarg):
-        unique_key = self._get_unique_key(task_args, task_kwarg)
+    def _set_ignore_task_after_success(self, task_args, task_kwargs):
+        ignore_task_after_success_key = self._get_ignore_task_after_success_key(task_args, task_kwargs)
+        if ignore_task_after_success_key:
+            cache.add(
+                ignore_task_after_success_key,
+                True,
+                (now() + self.ignore_task_after_success_timedelta - now()).total_seconds()
```

Comment:
Technically, you should save current time into a variable.

- you don't call function twice
- you are guaranteed to operate with same value

Reply:
good point